<a href="https://colab.research.google.com/github/srenukaa/flyrank-ml-assignments/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/srenukaa/flyrank-ml-assignments/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I am choosing the Content Refresh and Optimization Lane. In SEO and content performance management, identifying which specific pages are entering a clear downward trend before they lose all ranking equity is a critical challenge. Predicting content decay early allows the team to prioritize high-value pages that need immediate rewrite or structural updates, rather than guessing blindly across thousands of URLs

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The Decision: Which exact pages are truly declining in search performance and require engineering/editorial budget to refresh?

The Action: The content operations team shifts their focus to rewrite, update external/internal links, or optimize the target intent of the flagged pages.

Cost of a Wrong Call: >   * False Positive (Predicting decline when stable/growing): Waste of content team budget and human hours rewriting a page that was already performing perfectly fine.

False Negative (Missing a real decline): Leaving a core revenue-generating page to decay silently, leading to a permanent drop in organic traffic and conversions.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
# Clone the repository and change the working directory
!git clone https://github.com/srenukaa/flyrank-ml-assignments.git
import os

# Change to the notebook's directory within the cloned repo
# This assumes the notebook is in 'work/notebooks/' relative to the repo root
# and that the repo was cloned into '/content/flyrank-ml-assignments'
os.chdir('flyrank-ml-assignments/work/notebooks')

Cloning into 'flyrank-ml-assignments'...
remote: Enumerating objects: 127, done.
remote: Counting objects: 100% (127/127), done.
remote: Compressing objects: 100% (76/76), done.
remote: Total 127 (delta 43), reused 111 (delta 35), pack-reused 0 (from 0)
Receiving objects: 100% (127/127), 1.84 MiB | 12.15 MiB/s, done.
Resolving deltas: 100% (43/43), done.


In [ ]:
import pandas as pd

# Load the starter dataset
# The path has been adjusted to account for the notebook being in a subdirectory
# (e.g., 'work/notebooks/') relative to the repository root where 'data/' is located.
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# 1. Get total row count to establish database baseline
total_rows = len(df)

# 2. Get the baseline mean CTR for page_1 to see what's at stake
visible = df[df["impressions_90d"] >= 100]
page_1_ctr = visible.groupby("position_tier")["ctr"].mean().get("page_1", 0)

# 3. Check the directional correlation between rank position and CTR
pos_ctr_corr = df["avg_position"].corr(df["ctr"])

print(f"Total rows in dataset: {total_rows}")
print(f"Mean CTR for Page 1 elements: {page_1_ctr:.4f}")
print(f"Correlation between Average Position and CTR: {pos_ctr_corr:.3f}")

Total rows in dataset: 30000
Mean CTR for Page 1 elements: 0.3548
Correlation between Average Position and CTR: -0.073


The dataset provides a solid foundation of 30,000 rows to discover meaningful trends.

The steep drop-off is mathematically real: Page 1 average CTR sits at 35.48%, making rank protection vital.

There is an observed negative directional correlation of -0.073 between average position and CTR, confirming that as the position number falls (getting closer to rank #1), click-through performance systematically improves.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I CAN claim: The model provides directional decision-support. It uncovers observed patterns and risk signals within our specific dataset to highlight which content is structurally aligned with historical decay cycles.

What I CANNOT claim: This work does not establish causal proof, nor does it claim to "crack or predict Google's live ranking algorithm." It simply mirrors behavior observed in our historical operational metrics to guide resource allocation.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.